### PR TITLE
decoupled data source from config

### PR DESCRIPTION
**Pull request by greed2411**
tried decoupling the data source from `Config`, we may have to fix the rest of the code to support the tests to happen. but let me know if there is any need to make changes in the suggested classes and methods.

### CODE REVIEW

File: slu/slu/utils/config.py [hunk 1]
```diff
@@ -92,7 +142,8 @@ def __attrs_post_init__(self) -> None:
             const.CLASSIFICATION
         ][const.FORMAT]
 
-        self.rules = self._config[const.RULES]
+        # self.rules = self._config[const.RULES]
+        self.rules = self._config["postprocess"][0]["params"]["rules"]
```
Comment (ltbringer) [inline slu/slu/utils/config.py:146]: We can't guarantee that rules will always be at that position (always question when accessing list indices), it is better to handle this in a separate PR.

File: slu/tests/test_config/test_config_creation.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+def test_config_json():
+
+
+    from slu.utils.config import Config, JSONAPIConfigDataProvider
+
+    json_config_data_provider = JSONAPIConfigDataProvider()
+
+    config = Config(config_data_provider=json_config_data_provider)
+    # assert config._config ==
```
Comment (ltbringer) [inline slu/tests/test_config/test_config_creation.py:25]: Assertion is commented? can we mock it using httpretty? dialogy has examples.

**Review by ltbringer — CHANGES_REQUESTED** [review]

File: slu/slu/utils/config.py (hunk 1 above)
Comment (greed2411) [reply slu/slu/utils/config.py:146]: yeah yeah. but that's what it took to run and pass the tests with future axis config.

Comment (greed2411) [reply slu/slu/utils/config.py]: fixed it by reverting  to the old way itself, but if we don't fix it. none of the new `Config` would work. nor won't the tests.

File: slu/tests/test_config/test_config_creation.py (hunk 2 above)
Comment (greed2411) [reply slu/tests/test_config/test_config_creation.py]: done.

**Review by ltbringer — APPROVED** [review]

**Pull request merged**